### PR TITLE
fix invalid break continue return yield

### PR DESCRIPTION
**Pull request by AaronRocinante**
#497

### CODE REVIEW

File: vm/src/compile.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ struct Compiler {
     nxt_label: usize,
     source_path: Option<String>,
     current_source_location: ast::Location,
-    in_loop: bool,
+    loop_depth: usize,
```
Comment (OddCoincidence) [inline vm/src/compile.rs]: Did you see my comment [here](https://github.com/RustPython/RustPython/issues/497#issuecomment-466732095)? The `in_loop` flag should suffice.

File: vm/src/bytecode.rs [hunk 2]
```diff
@@ -252,6 +253,7 @@ impl CodeObject {
         source_path: String,
         first_line_number: usize,
         obj_name: String,
+        is_function_obj: bool,
```
Comment (OddCoincidence) [inline vm/src/bytecode.rs]: Why add this here instead of another flag in `Compiler` like `is_loop`?

File: vm/src/compile.rs (hunk 1 above)
Comment (AaronRocinante) [reply vm/src/compile.rs]: Changed.

File: vm/src/bytecode.rs (hunk 2 above)
Comment (AaronRocinante) [reply vm/src/bytecode.rs]: Both ways would work. I think this has the advantage that we don't have to remember to save and restore the flag when exiting the block. It's also how it's done in the cpython implementation.

Comment (OddCoincidence) [reply vm/src/bytecode.rs]: > Both ways would work. I think this has the advantage that we don't have to remember to save and restore the flag when exiting the block.

True but it has the disadvantage that it makes the information accessible in a wider scope than is currently necessary. So unless there is some feature that depends on distinguish code objects that are functions, I'd prefer just to use a flag for now.

> It's also how it's done in the cpython implementation.

Could you link to the relevant cpython bits?

Comment (AaronRocinante) [reply vm/src/bytecode.rs]: https://github.com/python/cpython/blob/master/Python/compile.c#L2744
https://github.com/python/cpython/blob/master/Python/compile.c#L2758

Comment (AaronRocinante) [reply vm/src/bytecode.rs]: @OddCoincidence I've changed to use boolean flags for checking the invalid return and yield.

**Comment by OddCoincidence** [comment]
Thanks!

**Pull request merged**